### ISSUE TITLE
The terraform-provider-google_v4.43.0_x5 plugin crashed!

### ISSUE THREAD

**Issue by klevinkona** [terraform-google-modules/terraform-google-network#402]
### TL;DR

Error: The terraform-provider-google_v4.43.0_x5 plugin crashed!
Started to test with the official google link
https://cloud.google.com/docs/terraform/get-started-with-terraform

### Observed behavior

╷
│ Error: Inconsistent dependency lock file
│
│ The following dependency selections recorded in the lock file are inconsistent with the current configuration:
│   - provider registry.terraform.io/hashicorp/random: required by this configuration but no version is selected
│
│ To update the locked dependency selections to match a changed configuration, run:
│   terraform init -upgrade
╵
root@cs-431513801194-default:/home/kona_klevin/tf-tutorial# terraform init -upgrade

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/random...
- Finding latest version of hashicorp/google...
- Installing hashicorp/random v3.4.3...
- Installed hashicorp/random v3.4.3 (signed by HashiCorp)
- Using previously-installed hashicorp/google v4.43.0

Terraform has made some changes to the provider dependency selections recorded
in the .terraform.lock.hcl file. Review those changes and commit them to your
version control system if they represent changes you intended to make.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
root@cs-431513801194-default:/home/kona_klevin/tf-tutorial# terraform plan
╷
│ Error: Invalid provider configuration
│
│ Provider "registry.terraform.io/hashicorp/google" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in the
│ provider documentation.
│
╵
╷
│ Error: Plugin did not respond
│
│   with provider["registry.terraform.io/hashicorp/google"],
│   on <empty> line 0:
│   (source code not available)
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-google_v4.43.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 10 [running]:
github.com/hashicorp/terraform-provider-google/google.GetCurrentUserEmail(0x32373f8?, {0xc00007a150?, 0x322e820?})
        github.com/hashicorp/terraform-provider-google/google/utils.go:511 +0x125
github.com/hashicorp/terraform-provider-google/google.(*Config).logGoogleIdentities(0xc00032d000)
        github.com/hashicorp/terraform-provider-google/google/config.go:589 +0xd4
github.com/hashicorp/terraform-provider-google/google.(*Config).LoadAndValidate(0xc00032d000, {0x32373f8?, 0xc0005c6540})
        github.com/hashicorp/terraform-provider-google/google/config.go:480 +0x265
github.com/hashicorp/terraform-provider-google/google.providerConfigure({0x32374a0, 0xc000c79f20}, 0x2c9e360?, 0x1?)
        github.com/hashicorp/terraform-provider-google/google/provider.go:1627 +0x2a05
github.com/hashicorp/terraform-provider-google/google.Provider.func1({0x32374a0?, 0xc000c79f20?}, 0xc000c79f20?)
        github.com/hashicorp/terraform-provider-google/google/provider.go:905 +0x27
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc000300cc0, {0x32374a0, 0xc000c79f20}, 0xd?)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/provider.go:297 +0x1e6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0000c3b60, {0x32374a0?, 0xc000c785a0?}, 0xc000b7e7c8)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:557 +0x345
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc000144500, {0x32374a0?, 0xc000c2ff80?}, 0xc0005c6140)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:556 +0x2ce
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x2c63920?, 0xc000144500}, {0x32374a0, 0xc000c2ff80}, 0xc000c53f10, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000001e0, {0x323c888, 0xc0002e4000}, 0xc0008dcc60, 0xc0009bcdb0, 0x438de18, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000001e0, {0x323c888, 0xc0002e4000}, 0xc0008dcc60, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-google_v4.43.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Terraform Configuration

```hcl
resource "google_compute_network" "vpc_network" {
  name                    = "my-custom-mode-network"
  auto_create_subnetworks = false
  mtu                     = 1460
}

resource "google_compute_subnetwork" "default" {
  name          = "my-custom-subnet"
  ip_cidr_range = "10.0.1.0/24"
  region        = "us-west1"
  network       = google_compute_network.vpc_network.id
}
```

### Terraform Version

```sh
Terraform v1.3.3
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.43.0
+ provider registry.terraform.io/hashicorp/random v3.4.3

Your version of Terraform is out of date! The latest version
is 1.3.4. You can update by downloading from https://www.terraform.io/downloads.html
```
